server/lib/api/address.js

```diff
@@ -1,38 +1,121 @@
-module.exports = function addressrouter(router) {
+const Block = require('../../models/block.js');
+const logger = require('../logger');
+
+const MAX_BLOCKS = 200;
+
+// Shoe horned in. Not dry, also in blocks. Make db api later
+function getBlock(params, options, limit, cb) {
+  const defaultOptions = { _id: 0 };
+
+  if (!Number.isInteger(limit)) {
+    limit = MAX_BLOCKS;
+  }
+
+  Object.assign(defaultOptions, options);
+
+  Block.find(
+    params,
+    defaultOptions,
+    cb)
+    .sort({ height: -1 })
+    .limit(limit);
+}
+
+module.exports = function AddressAPI(router) {
   router.get('/addr/:addr', (req, res) => {
-    res.send(req.params.addr);
+    res.json({
+      addrStr:                 req.params.addr,
+      balance:                 0,
+      balanceSat:              0,
+      totalReceived:           0,
+      totalReceivedSat:        0,
+      totalSent:               0,
+      totalSentSat:            0,
+      unconfirmedBalance:      0,
+      unconfirmedBalanceSat:   0,
+      unconfirmedTxApperances: 0,
+      txApperances:            5,
+    });
   });
 
   router.get('/addr/:addr/utxo', (req, res) => {
-    res.send(req.params.addr);
+    res.send('1');
   });
 
   router.get('/addr/:addr/balance', (req, res) => {
-    res.send(req.params.addr);
+    res.send('2');
   });
 
   router.get('/addr/:addr/totalReceived', (req, res) => {
-    res.send(req.params.addr);
+    res.send('3');
   });
 
   router.get('/addr/:addr/totalSent', (req, res) => {
-    res.send(req.params.addr);
+    res.send('4');
   });
 
   router.get('/addr/:addr/unconfirmedBalance', (req, res) => {
-    res.send(req.params.addr);
+    res.send('5');
   });
 
   router.get('/addrs/:addrs/utxo', (req, res) => {
-    res.send(req.params.addrs);
+    res.send('6');
   });
 
   router.post('/addrs/utxo', (req, res) => {
-    res.send('post stub');
+    res.send('7');
   });
 
   router.get('/addrs/:addrs/txs', (req, res) => {
-    res.send(req.params.addrs);
+    getBlock(
+      {
+        $or:
+        [
+          { 'txs.outputs.address': req.params.addr },
```

Comment:
this query style could work with the right supporting indexes. the problem would be that the results will be blocks, including all the transactions (not just the relevant ones for requested address). To accomplish this style of querying but only outputting the relevant address transactions would likely most eficient using an aggregation pipeline with $unwind.